### ISSUE TITLE
Feature Request - Recycle bin

### ISSUE THREAD

**Issue by mx4k** [docmost/docmost#59]
Hi, love your work so far. Please keep on going.

It would really be helpful, if you were able to have a recycle bin for pages. Or maybe it can be added to the page history of the workspace / parent page.

**Comment by sabirovrinat85** [comment]
Yep, that feature has a real necessity.

And that recycle bin should have it's own set of permissions per space, like one can make deleted pages removable only by workspace owner or members of some group, but restoring is accessible for those who has write access to a space.

PS: Small interesting feature would be that some deleted pages can be marked as invisible to others, still being in a trash bin

**Comment by windy20000** [comment]
Yes. I deleted the page unintentionally when I was using iPad. I cannot recover it.

**Comment by ahmadnugroho-asp** [comment]
i think it should have this feature.

**Comment by RubenRuCh** [comment]
Same here, I think this feature would be awesome to have

I created a [$100 bounty](https://app.opire.dev/issues/01J7N4KQP4J7G7K3W7KEPKHEJP) for this issue. Hope this can help to attract the attention of some developer that can tackle this

**Comment by mobley-trent** [comment]
I'd like to have a crack at this issue

**Comment by mobley-trent** [comment]
Hello @ahmadnugroho-asp , Are there any specific instructions on how to set up the postgres db locally ? I couldn't find them in the docs. I ran into this error while setting up the build. It occurred with the `workspaces` table first, but I don't want to get the schema wrong for the tables. 

![Screenshot 2024-09-16 151658](https://github.com/user-attachments/assets/aa4090af-e0ef-4584-a19e-342694cbe27a)

I tried booting up the image but the alpine linux container had a problem with installing git, so I couldn't proceed.

**Comment by Philipinho** [comment]
@mobley-trent you are in development mode. Please see the development guide https://docmost.com/docs/self-hosting/development#migrations.

You have to run the migrations manually.
```sh
pnpm nx run server:migration:latest
```

**Comment by mobley-trent** [comment]
Hi @Philipinho I'd appreciate your feedback on the feature

**Comment by Shu12388y** [comment]
@RubenRuCh @mx4k can you assign this issue to me

**Comment by mx4k** [comment]
@Shu12388y sure

**Comment by mx4k** [comment]
@krishanjangid https://github.com/docmost/docmost/pull/325
